crates/nu-cli/src/shell/completer.rs

```diff
@@ -28,6 +31,25 @@ impl NuCompleter {
             .map(|block| completion::engine::completion_location(line, &block.block, pos))
             .unwrap_or_default();
 
+        // let matcher = nu_data::config::config(Tag::unknown())
+        //     .ok()
+        //     .and_then(|cfg| cfg.get("line_editor").cloned())
+        //     .and_then(|le| le.row_entries().find(|(idx, _value)| idx.as_str() == "completion_match_method"))
+        //     .and_then(|(_idx, value)| value.as_string().ok())
+        //     .unwrap_or_else(|| "".to_string());
```

Comment:
So here's what I had to do to get this working:

```rust
         let matcher = nu_data::config::config(Tag::unknown())
             .ok()
             .and_then(|cfg| cfg.get("line_editor").cloned())
             .and_then(|le| {
                 le.row_entries()
                     .find(|(idx, _value)| idx.as_str() == "completion_match_method")
                     .and_then(|(_idx, value)| value.as_string().ok())
             })
             .unwrap_or_else(String::new);
```

A couple of things:

- A lot of things getting borrowed, hence needing to clone or do something to get ownership of a thing that you can move around. I decided to move the `as_string().ok()` into the block above it. An alternative would be to `.map(|(_idx, value)| value.clone())` with what you have.
- I switched from `|| "".to_string` to simply `String::new`. Both work, but I like the simplicity of `String::new`.